Argilla 2.4: Curate Hub Datasets with Human Feedback—No Code Needed (#2448)
huggingface/blog: argilla-ui-hub.md

---

> title: "Argilla 2.4: Curate Hub Datasets with Human Feedback—No Code Needed"

Reviewer: ```suggestion
title: "Argilla 2.4: Easily Add Human Feedback to Hub Datasets—No Code Required"
```

thumbnail: /blog/assets/argilla-ui-hub/thumbnail.png
authors:
- user: nataliaElv
- user: burtenshaw
- user: dvilasuero
---

> # Argilla 2.4: Curate Hub Datasets with Human Feedback—No Code Needed

Author: This title needs to be changed as well

> We are extremely excited to share a major new feature: you can now start off your AI dataset projects without code. Using Argilla’s UI, you can easily import any dataset from the Hugging Face Hub, define questions, and start collecting human feedback.

Reviewer: ```suggestion
We are incredibly excited to share the most impactful feature since Argilla joined Hugging Face: you can start your AI dataset projects without code and from any Hub dataset.

Using Argilla’s UI, you can easily import a dataset from the Hugging Face Hub, define questions, and start collecting human feedback.
```

> [!NOTE]

> > Not familiar with Argilla? Argilla is a free, open-source data curation tool. Using Argilla, AI developers and domain experts can collaborate and build high quality datasets. Argilla is part of the Hugging Face family and it’s fully integrated with the Hub: from deploying it on Spaces to fine-tuning models with AutoTrain and TRL. Want to know more? Here’s an [intro blog post](https://huggingface.co/blog/dvilasuero/argilla-2-0).

Reviewer: ```suggestion
> Not familiar with Argilla? Argilla is a free, open-source data-centric tool. Using Argilla, AI developers and domain experts can collaborate and build high-quality datasets. Argilla is part of the Hugging Face family and fully integrated with the Hub. Want to know more? Here’s an [[intro blog post](https://huggingface.co/blog/dvilasuero/argilla-2-0)](https://huggingface.co/blog/dvilasuero/argilla-2-0).
```

Author: Fix link

Why is this new feature important to you and the community?

> - There are 230k datasets on the Hugging Face hub that you can use as a foundation to your own AI project.

Reviewer: ```suggestion
- The Hugging Face hub contains 230k datasets you can use as a foundation for your AI project.
```

> - For dataset publishers, it simplifies the process of collecting human feedback from the whole Hugging Face community or specialized teams and communities.

Reviewer: ```suggestion
- It simplifies collecting human feedback from the Hugging Face community or specialized teams.
```

> - This makes the step of going from a dataset in the Hub to human feedback workflows easier and faster. 

Reviewer: remove this line

> - This empowers users that aren’t familiar with Python or code, allowing them to create datasets using a UI instead of code.

Reviewer: ```suggestion
- It democratizes dataset creation for users with extensive knowledge about a specific domain who are unsure about writing code.
```

## Use cases

> This new feature enables a new set of use cases for building high quality datasets on the Hub:

Reviewer: ```suggestion
This new feature democratizes building high-quality datasets on the Hub:
```

> - You have just published an open dataset and want the community to contribute: import it into a public Argilla Space and share the URL with the world!

Reviewer: ```suggestion
- If you are a dataset publisher and want the community to contribute, import it into a public Argilla Space and share the URL with the world!
```

Author: Don't you think "dataset publisher" sounds like a job title? I'm not sure how many people would feel represented by that.

Reviewer: ```suggestion
- If you have published an open dataset and want the community to contribute, import it into a public Argilla Space and share the URL with the world!
```

> - You want to start annotating a new dataset from scratch: upload a CSV to the Hub, import it into an Argilla Space and start labeling!

Reviewer: ```suggestion
- If you want to start annotating a new dataset from scratch, upload a CSV to the Hub, import it into your Argilla Space, and start labeling!
```

- You want to curate an existing Hub dataset for your use case or contribute to improve it: import it into an Argilla Space and start curating!

## How it works

> First, you need to deploy Argilla. The recommended way is to deploy on Spaces, [following this guide](https://docs.argilla.io/latest/getting_started/quickstart/). The default deployment comes with Hugging Face OAuth enabled, meaning your Space will be open to any Hub users to annotate your dataset. This is perfect for use cases when you want the community to contribute to your dataset. If you want to keep the annotation restricted to you and other collaborators, [check this other guide](https://docs.argilla.io/latest/getting_started/how-to-configure-argilla-on-huggingface/).

Reviewer: ```suggestion
First, you need to deploy Argilla. The recommended way is to deploy on Spaces [[following this guide](https://docs.argilla.io/latest/getting_started/quickstart/)](https://docs.argilla.io/latest/getting_started/quickstart/). The default deployment comes with Hugging Face OAuth enabled, meaning your Space will be open to any Hub users to annotate your dataset. OAuth is perfect for use cases when you want the community to contribute to your dataset if you're going to keep the annotation restricted to you and other collaborators,[ [check this other guide](https://docs.argilla.io/latest/getting_started/how-to-configure-argilla-on-huggingface/)](https://docs.argilla.io/latest/getting_started/how-to-configure-argilla-on-huggingface/).
```

Author: Have you checked if those links work like that in the preview?

[screen recording]

> Once Argilla is running, sign in and click the “Import dataset from Hugging Face" button in the Home page. You can get started with one of our example datasets or input the repo id of the dataset you want to use. Argilla will automatically suggest an initial configuration based on the dataset’s features, so you don’t need to start from scratch.

Reviewer: ```suggestion
Once Argilla runs, sign in and click the “Import dataset from Hugging Face” button on the Home page. You can start with one of our example datasets or input the repo id of the dataset you want to use. Argilla automatically suggests an initial configuration based on the dataset’s features, so you don’t need to start from scratch. 
```

Author: I think "is running" is more correct in this context. I'll apply the other changes

> [!NOTE]

> > In this first version, the Hub dataset must be public, if you are interested in support for private datasets, we’d love to hear from you on [GitHub](https://github.com/argilla-io/argilla).

Author: Semicolons are too formal as a punctuation mark. I'll change that for a full stop.
```suggestion
> In this first version, the Hub dataset must be public. If you are interested in support for private datasets, we’d love to hear from you on [GitHub](https://github.com/argilla-io/argilla).
```

> The dataset’s columns will be mapped to fields and questions in Argilla. Fields include the data that you want feedback on, like text, chats, or images. Questions are the feedback you want to collect, like labels, ratings, rankings, or text. If you need, you can add and configure questions or remove unnecessary fields. All of the changes that you make will be previewed in real time, so you can see how your changes affect the dataset.

Reviewer: ```suggestion
The goal is to map dataset columns to fields and questions in Argilla. Fields include the data you want feedback on, like text, chats, or images. Questions are the feedback you wish to collect, like labels, ratings, rankings, or text. You can add and configure questions or remove unnecessary fields if needed. You can preview all changes in real time to get a clear idea of the Argilla dataset you’re configuring.
```

> Once you’re happy with the result, you can click on “Create dataset” to import the dataset. Now you’re ready to give feedback!

Reviewer: ```suggestion
Once you’re happy with the result, click “Create dataset” to import the dataset. Now you’re ready to give feedback!
```

> This new workflow streamlines the import of datasets from the Hub, but you can still [import datasets using Argilla’s Python SDK](https://docs.argilla.io/latest/how_to_guides/dataset/) if you need further customization.

Reviewer: ```suggestion
You can try this for yourself by following the [quickstart guide](https://docs.argilla.io/latest/getting_started/quickstart/). It takes under 5 minutes!

This new workflow streamlines the import of datasets from the Hub, but you can still [[import datasets using Argilla’s Python SDK](https://docs.argilla.io/latest/how_to_guides/dataset/) ](https://docs.argilla.io/latest/how_to_guides/dataset/)if you need further customization.
```

Author: Same issue with the link here

> We’d love to know what you think or how your experience was, so we can make the import of Hub datasets even better. Let us know on GitHub or the HF Discord!

Reviewer: ```suggestion
We’d love to hear your thoughts and first experiences. Let us know on GitHub or the HF Discord!
```

> In this first version, the Hub dataset must be public. If you are interested in support for private datasets, we’d love to hear from you on [GitHub](https://github.com/argilla-io/argilla/issues/new/choose).

Argilla automatically suggests an initial configuration based on the dataset’s features, so you don’t need to start from scratch, but you can add questions or remove unnecessary fields. Fields should include the data you want feedback on, like text, chats, or images. Questions are the feedback you wish to collect, like labels, ratings, rankings, or text. All changes are shown in real time, so you can get a clear idea of the Argilla dataset you’re configuring. 

Once you’re happy with the result, click “Create dataset” to import the dataset with your configuration. Now you’re ready to give feedback!

You can try this for yourself by following the [quickstart guide](https://docs.argilla.io/latest/getting_started/quickstart/). It takes under 5 minutes!

This new workflow streamlines the import of datasets from the Hub, but you can still [import datasets using Argilla’s Python SDK](https://docs.argilla.io/latest/how_to_guides/dataset/) if you need further customization.

> We’d love to hear your thoughts and first experiences. Let us know on GitHub or the HF Discord!

Author: Done!

---

> title: "Argilla 2.4: Easily Add Human Feedback to Hub Datasets—No Code Required"

Author: Do you think this title matches well with the thumbnail @dvsrepo? I see a big semantic difference between "creating datasets" and "adding human feedback to datasets".

Reviewer: Yes, I think it matches well; both have different purposes and don't need to be aligned 100%

---

> title: "Argilla 2.4: Easily Add Human Feedback to Hub Datasets—No Code Required"

Author: ```suggestion
title: "Argilla 2.4: Easily Build Fine-Tuning and Evaluation Datasets on the Hub — No Code Required"
```

thumbnail: /blog/assets/argilla-ui-hub/thumbnail.png
authors:
- user: nataliaElv
- user: burtenshaw
- user: dvilasuero
---

> # Argilla 2.4: Easily Add Human Feedback to Hub Datasets—No Code Required

Reviewer: ```suggestion
# Argilla 2.4: Easily Build Fine-Tuning and Evaluation Datasets on the Hub — No Code Required
```

> We are incredibly excited to share the most impactful feature since Argilla joined Hugging Face: you can start your AI dataset projects without code and from any Hub dataset.

Reviewer 2: ```suggestion
We are incredibly excited to share the most impactful feature since Argilla joined Hugging Face: you can prepare your AI datasets without any code, getting started from any Hub dataset!
```

Not sure if we should add some more details here, like "getting started from any Hub dataset and enriching it with human feedback and annotations". The next sentence is about it too.

> Using Argilla’s UI, you can easily import a dataset from the Hugging Face Hub, define questions, and start collecting human feedback.

Reviewer 2: I'd maybe combine this with the previous one instead of having them as separate paragraphs.

Why is this new feature important to you and the community?

- The Hugging Face hub contains 230k datasets you can use as a foundation for your AI project.
- It simplifies collecting human feedback from the Hugging Face community or specialized teams.
- It democratizes dataset creation for users with extensive knowledge about a specific domain who are unsure about writing code.

## Use cases

This new feature democratizes building high-quality datasets on the Hub:

- If you have published an open dataset and want the community to contribute, import it into a public Argilla Space and share the URL with the world!
- If you want to start annotating a new dataset from scratch, upload a CSV to the Hub, import it into your Argilla Space, and start labeling!
- If you want to curate an existing Hub dataset for fine-tuning or evaluating your model,  import the dataset into an Argilla Space and start curating!

> 

Reviewer 2: ```suggestion
```

- If you want to improve an existing Hub dataset to benefit the community, import it into an Argilla Space and start giving feedback!

## How it works

> First, you need to deploy Argilla. The recommended way is to deploy on Spaces [following this guide](https://docs.argilla.io/latest/getting_started/quickstart/). The default deployment comes with Hugging Face OAuth enabled, meaning your Space will be open to any Hub users to annotate your dataset. OAuth is perfect for use cases when you want the community to contribute to your dataset if you're going to keep the annotation restricted to you and other collaborators,[check this other guide](https://docs.argilla.io/latest/getting_started/how-to-configure-argilla-on-huggingface/).

Reviewer 2: ```suggestion
First, you need to deploy Argilla. The recommended way is to deploy on Spaces [following this guide](https://docs.argilla.io/latest/getting_started/quickstart/). The default deployment comes with Hugging Face OAuth enabled, meaning your Space will be open for annotation contributions from any Hub user. OAuth is perfect for use cases when you want the community to contribute to your dataset. If you want to restrict annotation to you and other collaborators, [check this guide](https://docs.argilla.io/latest/getting_started/how-to-configure-argilla-on-huggingface/) for additional configuration options.
```

<video controls title="Import a HF dataset from the Hub in the Argilla UI">
<source src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/argilla-ui-hub/import_hub_dataset.mp4" type="video/mp4">
Import a HF dataset from the Hub in the Argilla UI
</video>

Once Argilla is running, sign in and click the “Import dataset from Hugging Face” button on the Home page. You can start with one of our example datasets or input the repo id of the dataset you want to use. Argilla automatically suggests an initial configuration based on the dataset’s features, so you don’t need to start from scratch.

> [!NOTE]

> > In this first version, the Hub dataset must be public. If you are interested in support for private datasets, we’d love to hear from you on [GitHub](https://github.com/argilla-io/argilla).

Reviewer 2: ```suggestion
> In this first version, you can import public datasets from the Hub. If you are interested in support for private datasets, we’d love to hear from you on [GitHub](https://github.com/argilla-io/argilla).
```

Maybe direct link to open an issue? https://github.com/argilla-io/argilla/issues/new/choose

> The goal is to map dataset columns to fields and questions in Argilla. Fields include the data you want feedback on, like text, chats, or images. Questions are the feedback you wish to collect, like labels, ratings, rankings, or text. You can add and configure questions or remove unnecessary fields if needed. You can preview all changes in real time to get a clear idea of the Argilla dataset you’re configuring.

Reviewer 2: ```suggestion
Before you can import the dataset, you have to map dataset columns to fields and questions in Argilla. Fields include the data you want feedback on, like text, chats, or images. Questions are the feedback you wish to collect, like labels, ratings, rankings, or text. You can add and configure questions or remove unnecessary fields if needed. You can preview all changes in real time to get a clear idea of the Argilla dataset you’re configuring.
```

> Once you’re happy with the result, click “Create dataset” to import the dataset. Now you’re ready to give feedback!

Reviewer 2: ```suggestion
Once you’re happy with the result, click “Create dataset” to import the dataset with your configuration. Now you’re ready to give feedback!
```
